### PR TITLE
Attempt to fix cmp e2e test flakiness

### PR DESCRIPTION
**Pull request by geoand**
**What is the purpose of this change? What does it change?**

This PR attempts to reduce the flakiness of the component end-to-end tests.
The source of flakiness that was identified and fixed with this PR was a race condition that was being incurred when updating a component a quick succession.

**Was the change discussed in an issue?**

No, but the team has been experiencing lots of flakiness the past few weeks

**How to test changes?**

This PR does not touch odo proper. All tests should now pass without flakiness

### CODE REVIEW

File: tests/e2e/cmp_test.go [hunk 1]
```diff
@@ -368,3 +378,15 @@ var _ = Describe("odoCmpE2e", func() {
...
 	})
 })
+
+// ensures that the DeploymentConfig of the specified component
+// has completely rolled out
+// this is very useful to avoid race conditions that can occur when
+// updating the component
+func waitForDCOfComponentToRollout(componentName string) {
+	dcName := runCmd(fmt.Sprintf("oc get dc -l app.kubernetes.io/component-name=%s -o name", componentName))
+	// oc rollout status ensures that the existing DC is fully rolled out before it teminates
```
Comment (metacosm) [inline tests/e2e/cmp_test.go]: typo: teminates -> terminates

File: tests/e2e/cmp_test.go [hunk 2]
```diff
@@ -368,3 +378,15 @@ var _ = Describe("odoCmpE2e", func() {
...
+	// we need this because a rolling DC could cause odo update to fail due to it's use
```
Comment (metacosm) [inline tests/e2e/cmp_test.go]: it's -> its :p

Comment (geoand) [reply tests/e2e/cmp_test.go]: Fixed, thanks!

File: tests/e2e/cmp_test.go (hunk 1 above)
Comment (geoand) [reply tests/e2e/cmp_test.go]: Done :)

**Comment by anmolbabu** [comment]
@geoand It seems the test once timed out

**Pull request merged**